libgpos/src/task/CAutoTaskProxy.cpp

```diff
@@ -506,18 +506,14 @@ CAutoTaskProxy::Execute
 	{
 		// mark task as erroneous
 		ptsk->SetStatus(CTask::EtsError);
-
-		if (m_fPropagateError)
-		{
-			GPOS_RETHROW(ex);
-		}
+		GPOS_RETHROW(ex);
 	}
 	GPOS_CATCH_END;
 
-	// reset exception context
+	// Raise exception if task encounters an exception
 	if (ptsk->FPendingExc())
 	{
-		ptsk->Perrctxt()->Reset();
+		GPOS_RETHROW(ptsk->Perrctxt()->Exc());
```

Comment:
Well, there, we should check whether we want to throw or not before we throw. Sorry, I am still trying to keep the original code path intact.